test/worker_test.rb

```diff
@@ -982,6 +980,31 @@ def self.perform
     assert_equal 0, messages.string.scan(/ERROR/).count
   end
 
+  class CounterJob
+    class << self
+      attr_accessor :perform_count
+    end
+    self.perform_count = 0
+
+    def self.perform
+      self.perform_count += 1
+    end
+  end
+
+  it "runs jobs without forking if fork isn't implemented" do
+    nil while @worker.reserve # empty queue
```

Comment:
Even if it is, the `before` block enqueues a job https://github.com/dylanahsmith/resque/blob/350d1b4306d0cc74aebc904e5d0e1e0a15a849d2/test/worker_test.rb#L22-L25